src/cpp/redisserver.cpp

```diff
@@ -80,32 +80,35 @@ std::string RedisServer::_get_ssdb()
     _check_ssdb_string(env_str);
 
     // Parse the data in it
-    std::vector<std::string> hosts_ports;
+    std::vector<SRAddress> address_choices;
     const char delim = ',';
 
     size_t i_pos = 0;
     size_t j_pos = env_str.find(delim);
     while (j_pos != std::string::npos) {
-        hosts_ports.push_back("tcp://"+
-        env_str.substr(i_pos, j_pos - i_pos));
+        std::string substr = env_str.substr(i_pos, j_pos - i_pos);
+        SRAddress addr_spec(substr);
+        address_choices.push_back(addr_spec);
         i_pos = j_pos + 1;
         j_pos = env_str.find(delim, i_pos);
     }
     // Catch the last value that does not have a trailing ';'
-    if (i_pos < env_str.size())
```

Comment:
Not caused by you, but the above comment says ';' is the separator -- can we update to ``,``

Reply:
done